core/autoscaling.go

```diff
@@ -639,9 +663,13 @@ func (a *autoScalingGroup) getInstanceState(instanceID *string) int64 {
 	return 48
 }
 
+func canTerminateInstance(instanceID *string, instanceState int64) bool {
+	return instanceID != nil && instanceState != 48 && instanceState != 32
```

Comment:
Please define the state numbers as constants instead of having them in-line.

Maybe they are already available from the SDK.